### PR TITLE
docs(react-infobutton): Adding information about why we are going with InfoIcon and InfoButton to the Spec

### LINKED ISSUE #28048: InfoButton: Update docs to mention decision on component variant and usage of these
Issue tracking the update of the docs relating the usage of InfoLabel, InfoTip, and InfoButton. 

Here's a summary of the decision we made:

There are two use cases we want to tackle, covering these means covering the 80-90%: 

1. When the bubble has a few words and no interactive elements.
2. When the bubble may have more than a few sentences and interactive elements.

From the discussion we settled on having two types of controls to solve this cases, the first one will be an InfoTip that will cover the first use case.

InfoTip
* An InfoTip is an icon/image that has a tooltip.
* InfoTips must keep the default cursor even when hovered.
* InfoTips are opened when hovered or focused.

For the second use case we will use InfoButton.

InfoButton
* An InfoButton is a button that has a popover attached to it.
* InfoButtons must use the pointer cursor to show this is a button.
* InfoButtons do not open when focused/hovered, instead they open on click/space/enter.
* When dismissing the bubble of an InfoButton, this can be done by pressing Escape, clicking outside the bubble or clicking the button again.
* InfoButton has role note and at first focuses on the popover surface.

Outstanding challenge (5-10%) would be using it with touch devices, it will be hard to differentiate between an InfoButton and an InfoTip.

### PR DESCRIPTION
**Pull request by sopranopillow**
This PR adds information on what our decision to go with InfoButton and InfoIcon is. It also adds a description of what is expected of each control.

## Related Issue(s)

- Fixes #28048

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/InfoButton-mount_base.html">10</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/InfoButton-mount.html">14</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/Avatar-mount_base.html">585</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/Avatar-mount.html">573</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/Button-mount_base.html">285</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/Button-mount.html">290</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/Field-mount_base.html">1019</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/Field-mount.html">1056</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/FluentProvider-mount_base.html">663</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/FluentProvider-mount.html">664</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/FluentProviderWithTheme-mount_base.html">80</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/FluentProviderWithTheme-mount.html">84</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">60</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">70</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">75</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">88</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/InfoButton-mount_base.html">10</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/InfoButton-mount.html">14</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/MakeStyles-mount_base.html">860</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/MakeStyles-mount.html">837</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/Persona-mount_base.html">1634</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/Persona-mount.html">1641</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/SpinButton-mount_base.html">1298</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28186/perf-test-react-components/SpinButton-mount.html">1284</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `64.935 kB`<br />`17.918 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `211.151 kB`<br />`58.715 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `36.168 kB`<br />`11.964 kB` |
| <samp>react-infobutton</samp> <br /> <abbr title='bundle-size/InfoButton.fixture.js'>InfoButton</abbr>                                                                        | `132.502 kB`<br />`40.405 kB` |
| <samp>react-infobutton</samp> <br /> <abbr title='bundle-size/InfoLabel.fixture.js'>InfoLabel</abbr>                                                                          | `135.967 kB`<br />`41.466 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.473 kB`<br />`2.196 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/c0214bcb7d9369db75eb842adf4253893bc0e160'>c0214bcb7d9369db75eb842adf4253893bc0e160</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-infobutton/docs/Spec.md [hunk 1]
```diff
@@ -116,3 +116,31 @@ There's no migration guide as `v0` and `v8` do not have an InfoButton component.
...
+- InfoIcons must keep the default cursor even when hovered.
+- InfoIcons are opened when hovered or focused.
+- InfoIcons do not move focus when opened.
+
+### InfoButton - InfoButton with a Popover
+
+- An InfoButton is a button that has a Popover attached to it.
+- InfoButtons must use the pointer cursor to show this is a button and will have a backplate to differenciate the icon vs button.
+- InfoButtons do not open when focused/hovered, instead they open on click/space/enter.
+- InfoButton's bubble can be dismissed by pressing Escape, clicking outside the bubble, or clicking the button again.
```
Comment (khmakoto) [inline packages/react-components/react-infobutton/docs/Spec.md:141]: It can also be dismissed just by moving focus to another item, right?

**Review by khmakoto — APPROVED** [review]

File: packages/react-components/react-infobutton/docs/Spec.md (hunk 1 above)
Comment (sopranopillow) [reply packages/react-components/react-infobutton/docs/Spec.md:141]: Correct, I'll add it

**Pull request merged**